dockerfiles/sentinel-harness/terratest.Dockerfile

```diff
@@ -0,0 +1,45 @@
+ARG SENTINEL_HARNESS_BASE_IMAGE_TAG=latest
+ARG SENTINEL_HARNESS_BASE_IMAGE_REPO=ghcr.io/pluralsh/sentinel-harness-base
+ARG SENTINEL_HARNESS_BASE_IMAGE=$SENTINEL_HARNESS_BASE_IMAGE_REPO:$SENTINEL_HARNESS_BASE_IMAGE_TAG
+
+ARG GO_VERSION=1.25
+
+# Use sentinel harness base image
+FROM ${SENTINEL_HARNESS_BASE_IMAGE} AS harness
+
+FROM golang:${GO_VERSION}-alpine AS final
+
+# Define build arguments for multi-arch support
+ARG TARGETOS
+ARG TARGETARCH
+ARG VERSION
+
+ENV CGO_ENABLED=0 \
+    GOOS=${TARGETOS} \
+    GOARCH=${TARGETARCH} \
+    GOCACHE=/sentinel/.cache
+
+# Create directories and fix permissions
+RUN mkdir -p /sentinel/.cache && chown -R 65532:65532 /sentinel
+
+# Install curl and certificates for downloading kubectl
+RUN apk add --no-cache curl ca-certificates
+
+# Install runtime dependencies + kubectl
+RUN apk add --no-cache curl ca-certificates && \
+    KUBECTL_VERSION=$(curl -L -s https://dl.k8s.io/release/stable.txt) && \
+    curl -L -o /usr/local/bin/kubectl \
+      "https://dl.k8s.io/release/${KUBECTL_VERSION}/bin/${TARGETOS}/${TARGETARCH}/kubectl" && \
+    chmod +x /usr/local/bin/kubectl
+
+COPY --from=harness /sentinel-harness /usr/local/bin/sentinel-harness
+# Change ownership of the harness binary to UID/GID 65532
+RUN chown -R 65532:65532 /usr/local/bin/sentinel-harness
+COPY dockerfiles/sentinel-harness/terratest /sentinel
```

Comment:
We should refactor this so that practically all of this dockerfile is in the base image, and users only need to extend it and add this copy command.  

Then the documentation is super simple and the lift to build their own tests is straightforward, if they have a lot of docker maintenance it's going to be a pain to adopt.

Reply:
done